test/instances/susde/SUSDe.t.sol

```diff
@@ -0,0 +1,736 @@
+        // value.
+        assertApproxEqAbs(
+            baseProceeds,
+            withdrawalShares.mulDown(lpSharePrice),
+            1e9
+        );
+    }
+
+    /// Long ///
+
+    function test_open_long_nonpayable() external {
+        vm.startPrank(bob);
+
+        // Ensure that sending ETH to `openLong` fails.
+        vm.expectRevert(IHyperdrive.NotPayable.selector);
+        hyperdrive.openLong{ value: 2e18 }(
+            1e18,
+            0,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: true,
+                extraData: new bytes(0)
+            })
+        );
+
+        // Ensure that sending ETH to `openShort` fails.
+        vm.expectRevert(IHyperdrive.NotPayable.selector);
+        hyperdrive.openLong{ value: 0.5e18 }(
+            1e18,
+            0,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: false,
+                extraData: new bytes(0)
+            })
+        );
+    }
+
+    function test_round_trip_long_instantaneous() external {
+        // Bob opens a long with base.
+        uint256 basePaid = hyperdrive.calculateMaxLong();
+        IERC20(hyperdrive.baseToken()).approve(address(hyperdrive), basePaid);
+        (uint256 maturityTime, uint256 longAmount) = openLong(bob, basePaid);
+
+        // Get some balance information before the withdrawal.
+        (
+            uint256 totalSupplyAssetsBefore,
+            uint256 totalSupplySharesBefore
+        ) = getSupply();
+        AccountBalances memory bobBalancesBefore = getAccountBalances(bob);
+        AccountBalances memory hyperdriveBalancesBefore = getAccountBalances(
+            address(hyperdrive)
+        );
+
+        // Bob closes his long with vault shares as the target asset.
+        uint256 shareProceeds = closeLong(bob, maturityTime, longAmount, false);
+        uint256 baseProceeds = convertToBase(shareProceeds);
+
+        // Bob should receive approximately as much base as he paid since no
+        // time as passed and the fees are zero.
+        assertApproxEqAbs(baseProceeds, basePaid, 1e9);
+
+        // Ensure that the withdrawal was processed as expected.
+        verifyWithdrawal(
+            bob,
+            baseProceeds,
+            false,
+            totalSupplyAssetsBefore,
+            totalSupplySharesBefore,
+            bobBalancesBefore,
+            hyperdriveBalancesBefore
+        );
+    }
+
+    function test_round_trip_long_maturity() external {
+        // Bob opens a long with base.
+        uint256 basePaid = hyperdrive.calculateMaxLong();
+        IERC20(hyperdrive.baseToken()).approve(address(hyperdrive), basePaid);
+        (uint256 maturityTime, uint256 longAmount) = openLong(bob, basePaid);
+
+        // Advance the time and accrue a large amount of interest.
+        advanceTime(POSITION_DURATION, 137.123423e18);
```

Comment:
is there a specific reason you chose this variable interest rate?

Reply:
No. I used this so that we were testing with a weird and large number. That said, I'll switch this to be a fuzz test since this is arbitrary, and it would be better to have a bound that we can actually take to the bank.